### PR TITLE
CWS: remove recursion in `FileActivityNode.InsertFileEvent`

### PR DESCRIPTION
**Pull request by paulcacheux**
### What does this PR do?

This PR removes the recursion in `FileActivityNode.InsertFileEvent`, protecting us a bit against stack overflows in case of a very long path

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/security/probe/activity_dump.go [hunk 1]
```diff
@@ -1359,30 +1358,45 @@ func (fan *FileActivityNode) enrichFromEvent(event *Event) {
...
+		// create new child
+		somethingChanged = true
+		if len(currentPath) <= nextParentIndex+1 {
+			currentFan.Children[parent] = NewFileActivityNode(fileEvent, event, parent, generationType)
+			break
+		} else {
+			child := NewFileActivityNode(nil, nil, parent, generationType)
+			currentFan.Children[parent] = child
+
+			somethingChanged = true
```
Comment (Gui774ume) [inline pkg/security/probe/activity_dump.go:1392]: Already set on line 1384 no ?

**Review by Gui774ume — APPROVED** [review]
I know we didn't have any, but we should really add unit tests with a hard coded tree and a few hard coded events that you insert in a new tree to check against the first one. LGTM but it's hard to know for sure without tests :(

**Comment by paulcacheux** [comment]
Absolutely agree on unit tests

**Comment by paulcacheux** [comment]
Here is https://github.com/DataDog/datadog-agent/pull/12900 for a start of unit tests

**Pull request merged**